packages/perseus/src/widgets/interactive-graphs/graphs/segment.tsx

```diff
@@ -54,46 +59,52 @@ const SegmentGraph = ({dispatch, graphState}: SegmentProps) => {
     }
 
     return (
-        <g
-            aria-label={wholeSegmentAriaLabel}
-            aria-describedby="segment-description"
-        >
+        <>
             {segments?.map((segment, i) => (
-                <MovableLine
+                <g
+                    aria-label={wholeSegmentAriaLabel}
+                    aria-describedby={`segment-description-${i}`}
                     key={i}
-                    points={segment}
-                    onMoveLine={(delta: vec.Vector2) => {
-                        dispatch(actions.segment.moveLine(i, delta));
-                    }}
-                    onMovePoint={(
-                        endpointIndex: number,
-                        destination: vec.Vector2,
-                    ) => {
-                        dispatch(
-                            actions.segment.movePointInFigure(
-                                i,
-                                endpointIndex,
-                                destination,
+                >
+                    <MovableLine
+                        key={i}
+                        points={segment}
+                        onMoveLine={(delta: vec.Vector2) => {
+                            dispatch(actions.segment.moveLine(i, delta));
+                        }}
+                        onMovePoint={(
+                            endpointIndex: number,
+                            destination: vec.Vector2,
+                        ) => {
+                            dispatch(
+                                actions.segment.movePointInFigure(
+                                    i,
+                                    endpointIndex,
+                                    destination,
+                                ),
+                            );
+                        }}
+                        ariaLabels={{
+                            point1AriaLabel: formatSegment(
+                                1,
+                                segment[0][X],
+                                segment[0][Y],
+                            ),
+                            point2AriaLabel: formatSegment(
+                                2,
+                                segment[1][X],
+                                segment[1][Y],
                             ),
-                        );
-                    }}
-                    ariaLabels={{
-                        point1AriaLabel: formatSegment(
-                            1,
-                            segment[0][X],
-                            segment[0][Y],
-                        ),
-                        point2AriaLabel: formatSegment(
-                            2,
-                            segment[1][X],
-                            segment[1][Y],
-                        ),
-                    }}
-                />
+                        }}
+                    />
+                    <g
+                        id={`segment-description-${i}`}
```

Comment:
You still need to use `React.useId()` to make sure that it's referring to the correct description if there are multiple interactive graphs on the page.

Reply:
Oh I see, you're talking about having multiple instances of the line segment graph on a single page, right? 
I wonder if the ID creation should be bubbled up to the Mafs graph to ensure that we're not getting the same ID in multiple instances for all the graphs, then the individual graphs can consume the IDs. 
It looks like we already have a [unique ID being generated there](https://github.com/Khan/perseus/blob/main/packages/perseus/src/widgets/interactive-graphs/mafs-graph.tsx#L95), we would just need to consume it

Reply:
You can bubble it down, but I think just adding another `React.useId()` at the top of the current file is much less work. Then you don't have to bubble it down into everything and keep passing it around everywhere even when not strictly necessary.

I'm not aware of performance issues with `React.useId()` so I'd personally go with that approach myself, but it's up to you.